keylime/user_utils.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+
+'''
+SPDX-License-Identifier: Apache-2.0
+Copyright 2017 Massachusetts Institute of Technology.
+'''
+
+import os
+import grp
+import pwd
+
+from keylime import keylime_logging
+
+# Configure logger
+logger = keylime_logging.init_logging('privileges')
+
+
+def string_to_uidgid(userandgroup):
+    """ Translate the userandgroup string to uid and gid.
+        The userandgroup parameter must be a string of the format '<user>[:<group>]'.
+        User and group can be strings or integers. If no group is given, -1 will be
+        returned for gid.
+    """
+    params = userandgroup.split(':')
+    if len(params) > 2:
+        return None, None, "User and group '%s' are in wrong format. Expected <user>[:group]" % userandgroup
+
+    gid = -1
+    if len(params) == 2:
+        if params[1].isnumeric():
+            gid = int(params[1])
+        else:
+            try:
+                gr = grp.getgrnam(params[1])
+                gid = gr.gr_gid
+            except Exception as e:
```

Comment:
Try to catch only exceptions that are expected. In this example:
```suggestion
            except KeyError as e:
```

Reply:
Fixed.

Reply:
Fixed.